archai/nlp/nvidia_transformer_xl/models/hf_gpt2.py

```diff
@@ -35,13 +34,15 @@ def __init__(self, **kwargs) -> None:
         if kwargs['d_embed'] == -1:
             kwargs['d_embed'] = kwargs['d_model']
 
-        assert all(elem == kwargs['n_head'][0] for elem in kwargs['n_head']), 'GPT2 does not accept heterogenous arch.'
```

Comment:
We still need these asserts, right?

Reply:
I would say not anymore because we will only be using the lists' first index at lines 43-45, so regardless of the list being heterogeneous or homogenous the model will be instantiated with its first value